src/vcpkg/commands.env.cpp

```diff
@@ -105,7 +105,12 @@ namespace vcpkg::Commands::Env
 #endif // ^^^ !_WIN32
         }
 
+#if defined(_WIN32)
         Command cmd("cmd");
+#else  // ^^^ _WIN32 / !_WIN32 vvv
+        Command cmd("");
+        Checks::exit_with_message(VCPKG_LINE_INFO, "Build environment commands are not supported on this platform");
```

Comment:
Can we implement it in non-Windows instead of directly disable it?

Reply:
I think it is two step, first step, correct error message, and explicitly says platform does not support

Second step, implement in Linux platform (in another issue 😉)

Reply:
@JackBoosY  if you agree with me, resolve this, and merge the PR, after that try for implementing env for each platform, (I think it is little big issue ;) )

Reply:
This needs to be localized.

Reply:
Another exit_with_message in this file exist without localization, Did you want to change all of them?

Reply:
@BillyONeal I localized message.

Please review again

Reply:
Yeah we need to fix those but it's not necessary for you to do that; we just wanted to 'put down the shovel' on that.